src/python_framework/network_dl.py

```diff
@@ -0,0 +1,17 @@
+from tqdm import tqdm
+import requests
+
+def download(fullpath, link):
+    output_to_file = "{fullpath}".format(fullpath=fullpath)
+    with requests.get(link, stream=True) as req:
+        total_size = int(req.headers.get('content-length', 0))
+        block_size = 1024 
+        t = tqdm(total = total_size, unit = 'iB', unit_scale = True, unit_divisor = block_size)
```

Comment:
Did you test this?  I've never used it, just stumbled across it several times...

Reply:
Yep -- tested and works just fine. I had to modify things a bit from some of the examples I saw and there are probably additional refinements (one that I left in a new `TODO` in the file) but I think it was exactly what we needed for this application. I also added the tqdm to the listed dependencies in the main readme.